arithmetization/src/test/java/net/consensys/linea/zktracer/SignedOperationsExtensiveTest.java

```diff
@@ -0,0 +1,110 @@
+
+import net.consensys.linea.testing.BytecodeCompiler;
+import net.consensys.linea.testing.BytecodeRunner;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class SignedOperationsExtensiveTest {
+  // See https://github.com/Consensys/linea-tracer/issues/1182 for documentation
+
+  static final String ZERO = "00".repeat(32);
+  static final String ONE = "00".repeat(31) + "01";
+
+  // 10^3 < SMALL_1 < SMALL_2 < 10^9
+  static final String SMALL_1 = "66" + randomBytes(1, 1);
+  static final String SMALL_2 = "66" + randomBytes(2, 2);
+
+  static final String LARGE_1 = randomBytes(16, 3);
+  static final String LARGE_2 = "01" + randomBytes(16, 4);
+
+  static final String RND_POS = "7f" + randomBytes(31, 5); // < 0x80 ...
+  static final String RND_NEG = "81" + randomBytes(31, 6); // > 0x80 ...
+
+  static final String NEG_ONE = "ff".repeat(32);
+
+  static String[] VALUES = {
+    ZERO, ONE, SMALL_1, SMALL_2, LARGE_1, LARGE_2, RND_POS, RND_NEG, NEG_ONE
+  };
+
+  @ParameterizedTest
+  @MethodSource("signedComparisonsModDivTestSource")
+  void signedComparisonsModDivTest(OpCode opCode, String a, String b) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram().push(b).push(a).op(opCode);
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(program.compile());
+    bytecodeRunner.run();
+  }
+
+  static Stream<Arguments> signedComparisonsModDivTestSource() {
+    List<Arguments> arguments = new ArrayList<>();
+    for (OpCode opCode : List.of(OpCode.SLT, OpCode.SGT, OpCode.SMOD, OpCode.SDIV)) {
+      for (String a : VALUES) {
+        for (String b : VALUES) {
+          arguments.add(Arguments.of(opCode, a, b));
+        }
+      }
+    }
+    return arguments.stream();
+  }
+
+  @ParameterizedTest
+  @MethodSource("signExtendTestSource")
+  void signExtendTest(String b, int x) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram().push(x).push(b).op(OpCode.SIGNEXTEND);
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(program.compile());
+    bytecodeRunner.run();
+  }
+
+  static Stream<Arguments> signExtendTestSource() {
+    List<Arguments> arguments = new ArrayList<>();
+    for (String firstByte : List.of("00", "ff", "7f", "81", "ff")) {
+      for (int nTrailingBytes : List.of(0, 15, 16, 30, 31)) {
+        int seed = firstByte.hashCode() * (nTrailingBytes + 1);
+        // This is just a way generate a different seed for each combination deterministically
+        String b = firstByte + randomBytes(nTrailingBytes, seed);
+        // In this case the most significant byte of b represents the sign
+        arguments.add(Arguments.of(b, b.length() / 2 - 1));
+        // In this other case we assume there is always 00 as a prefix
+        if (nTrailingBytes != 31) {
+          arguments.add(Arguments.of(b, b.length() / 2));
```

Comment:
A good source of inputs would be e.g.
```rusts
pos = 0, 1, 2, ..., 31, 32, 0xff,
      256 ^ 16 - 1, 256 ^ 16, 256 ^ 16 + 1,
      256 ^ 32 - 1
```
And
```rust
value = 0x 11 11 .. 11 [B] ff ff .. ff
```
Where `[B]` is a byte from among e.g. `{ 0x00, 0x56, 0x7f, 0x80, 0xc2, 0xff }` and occupies the position `i`-th byte, `i = 0, 1, 2, ..., 31`.

Reply:
Which would amount to `38 * 6 * 32 = 7 296` test cases, which is ok and which we will run on a weekly basis.

Reply:
@lorenzogentile404 `SIGNEXTEND` takes two arguments, one being the integer you want to sign extend, one being the position of the byte containing the sign bit.